contracts/test/spaces/membership/unit/MembershipJoinSpace.t.sol

```diff
@@ -518,4 +518,40 @@ contract MembershipJoinSpace is
     assertEq(withdrawAddress.balance, 0);
     assertEq(address(membership).balance, 0);
   }
+
+  function test_joinSpacePriceChangesMidTransaction()
+    external
+    givenMembershipHasPrice
+  {
+    vm.deal(bob, MEMBERSHIP_PRICE);
+
+    assertEq(membership.balanceOf(bob), 0);
+
+    vm.recordLogs();
+    vm.prank(bob);
+    membership.joinSpace{value: MEMBERSHIP_PRICE}(bob);
+    Vm.Log[] memory logs = vm.getRecordedLogs();
+
+    vm.prank(founder);
+    membership.setMembershipPrice(MEMBERSHIP_PRICE * 2);
+
+    (
+      address contractAddress,
+      bytes32 transactionId,
+      uint256 roleId,
+      address[] memory selectedNodes
+    ) = _getRequestedEntitlementData(logs);
+
+    for (uint256 i = 0; i < 3; i++) {
+      vm.prank(selectedNodes[i]);
+      IEntitlementGated(contractAddress).postEntitlementCheckResult(
+        transactionId,
+        roleId,
+        IEntitlementGatedBase.NodeVoteStatus.PASSED
+      );
+    }
+
+    assertEq(membership.balanceOf(bob), 1);
```

Comment:
I'm not sure if the above changes + tests fully address a comment Brian made about executing the example transaction in tenderly, where he said "in this specific case, the user had enough value and we should have proceeded and then refunded the balance." Do we have any testing that validates that if a user sends >> value than membership price they get a mint + a refund?

Reply:
don't think we should do refund and mint, you should be sending the exact amount

Reply:
I don't believe we can send exact amount, I believe there's always going to be a slippage window. Happy to talk about it.

Reply:
in Brian's use case the person didn't send more, they sent what was priced but the price dropped

Reply:
can you explain how this would happen with this fix implemented?